examples/sparse_gamma_def.py

```diff
@@ -0,0 +1,153 @@
+# this is an implementation of the sparse gamma deep exponential family experiment presented in
+# Ranganath, Rajesh, Tang, Linpeng, Charlin, Laurent, and Blei, David. Deep exponential families.
+#
+# the Olivetti faces dataset is originally from http://www.cl.cam.ac.uk/research/dtg/attarchive/facedatabase.html
+#
+# compare to the implementation here:
+# https://github.com/blei-lab/ars-reparameterization/tree/master/sparse%20gamma%20def
+
+
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import os
+
+import numpy as np
+import torch
+from torch import nn as nn
+
+import pyro
+import pyro.optim as optim
+import pyro.poutine as poutine
+import wget
+from pyro.distributions import Gamma, Poisson
+from pyro.infer import SVI, Trace_ELBO
+
+torch.set_default_tensor_type('torch.FloatTensor')
+pyro.enable_validation(True)
+pyro.util.set_rng_seed(0)
+
+
+class SparseGammaDEF(object):
+    def __init__(self):
+        # define the sizes of the layers in the deep exponential family
+        self.top_width = 100
+        self.mid_width = 40
+        self.bottom_width = 15
+        self.image_size = 64 * 64
+        # define hyparameters that control the prior
+        self.alpha_z = torch.tensor(0.1)
+        self.beta_z = torch.tensor(0.1)
+        self.alpha_w = torch.tensor(0.1)
+        self.beta_w = torch.tensor(0.3)
+        # define parameters used to initialize variational parameters
+        self.alpha_init = 0.5
+        self.mean_init = 0.0
+        self.sigma_init = 0.1
+        self.softplus = nn.Softplus()
```

Comment:
nit: if youre only using nn for softplus, theres a `torch.softplus` now which (i believe) is the same thing

Reply:
uh thanks